### PR TITLE
fixing frontend and backend services for user 

### PR DESCRIPTION
**Pull request by parth-agrawal**
- **fix controllers on backend**
- **fix api, services, and interfaces on frontend**
- **cleanup**

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 4c14740ee3784e160b92b02146024e85b8e13cf8  | 
|--------|--------|

### Summary:
Refactored backend user controller to use `UserService`, updated frontend services, and added Twitter authentication, including removal of unused backend types.

**Key points**:
- **Backend**: Refactored `backend/lib/controllers/users/controller.ts` to use `UserService` for CRUD operations.
- **Backend**: Removed `backend/lib/controllers/users/types.ts` and `backend/lib/services/user.ts`.
- **Frontend**: Updated `frontend/network/api.ts` for user endpoints.
- **Frontend**: Refactored `frontend/src/lib/services/users/service.ts` for CRUD operations.
- **Frontend**: Added `frontend/src/components/compound/SignInbutton.tsx` for Twitter authentication.
- **Frontend**: Updated `frontend/src/components/compound/CreateSpaceListingArea.tsx` to use `UserService`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: backend/lib/controllers/users/controller.ts [hunk 1]
```diff
@@ -70,20 +32,29 @@ router.post('/users/new', async (req, res) => {
...
+  const updatedUser = req.body;
   try {
-    const user = await client.user.update({
-      where: { clerkId: req.auth.userId },
-      data: {
-        email: req.body.email,
-        name: req.body.name,
-      },
-    });
+    const user = await UserService().updateUser({ updatedUser });
```
[the remark this answers is not in the document]
Comment (Hendersonajardimj) [reply backend/lib/controllers/users/controller.ts:39]: this actually seems true.. can't beside having not reviewed the userService function

File: backend/lib/controllers/users/controller.ts [hunk 2]
```diff
@@ -70,20 +32,29 @@ router.post('/users/new', async (req, res) => {
...
   } catch (error) {
     console.error("Error updating user:", error);
     res.status(500).json({ error: "Internal server error" });
   }
 });
 
+router.delete("/users/:userId", async (req, res) => {
+  const userId = req.params.userId;
+  try {
+    const user = await UserService().deleteUser({ userId });
```
[the remark this answers is not in the document]
Comment (Hendersonajardimj) [reply backend/lib/controllers/users/controller.ts:50]: same thing but could be totally off base

File: frontend/src/lib/services/users/types.ts
Comment (Hendersonajardimj) [inline frontend/src/lib/services/users/types.ts:1]: confused what this is or if you meant to add this but that's cause I don't rly know what is going on here

**Review by Hendersonajardimj — COMMENTED** [review]
Left a couple comments where I was confused

File: backend/lib/controllers/users/controller.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (parth-agrawal) [reply backend/lib/controllers/users/controller.ts:39]: i think this is incorrect - I extract the id from the updatedUser object itself, I don't need to pass it separately. Also I'm pretty sure I tested these endpoints using curl requests and it works correctly

File: backend/lib/controllers/users/controller.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (parth-agrawal) [reply backend/lib/controllers/users/controller.ts:50]: this is incorrect, deleteUser needs to take an object called {userId}. I also tested this endpoint and it works correctly

File: frontend/src/lib/services/users/types.ts
Comment (parth-agrawal) [reply frontend/src/lib/services/users/types.ts:1]: yup, i think this file should be deleted

**Pull request merged**